programs/fileio_asyncio.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) Yann Collet, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifndef ZSTD_FILEIO_ASYNCIO_H
+#define ZSTD_FILEIO_ASYNCIO_H
+
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+#include "../lib/common/mem.h"     /* U32, U64 */
+#include "fileio_types.h"
+#include "platform.h"
+#include "util.h"
+#include "../lib/common/pool.h"
+#include "../lib/common/threading.h"
+
+#define MAX_IO_JOBS          (10)
+
+typedef struct {
+    /* These struct fields should be set only on creation and not changed afterwards */
+    POOL_ctx* threadPool;
+    int totalIoJobs;
+    FIO_prefs_t* prefs;
+    POOL_function poolFunction;
+
+    /* Controls the file we currently write to, make changes only by using provided utility functions */
+    FILE* file;
+
+    /* The jobs and availableJobsCount fields are accessed by both the main and worker threads and should
+     * only be mutated after locking the mutex */
+    ZSTD_pthread_mutex_t ioJobsMutex;
+    void* availableJobs[MAX_IO_JOBS];
+    int availableJobsCount;
+} io_pool_ctx_t;
```

Comment:
A coding convention we try to follow in this code base : 
- variables and function names start with a lowercase letter (excluding the PREFIX_)
- type names start with an Uppercase letter

Reply:
Thank you @Cyan4973 , I went through the codebase and have seen multiple different conventions for type names (including in `fileio.c`), so I am a bit confused.
Would `IOPoolCtx_t` be good in this case?

Reply:
Indeed, enforcing a coding style is a thankless job, and can be difficult when there are many contributions.
So, in some instances, conventions may not be properly followed.
This should not stop us from trying to converge onto a consistent set of conventions.

> Would `IOPoolCtx_t` be good in this case?

Yes